tests/Http/HttpClientFactoryTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+
+namespace Http;
+
+
+use Microsoft\Graph\Core\NationalCloud;
+use Microsoft\Graph\Exception\ClientInitialisationException;
+use Microsoft\Graph\Http\HttpClientFactory;
+use Microsoft\Graph\Http\HttpClientInterface;
+
+class HttpClientFactoryTest extends \PHPUnit\Framework\TestCase
+{
+    function testNationalCloudWithEmptyString() {
+        $this->expectException(ClientInitialisationException::class);
+        $clientFactory = (new HttpClientFactory())->nationalCloud("");
+    }
+
+    function testNationalCloudWithInvalidUrl() {
+        $this->expectException(ClientInitialisationException::class);
+        $clientFactory = (new HttpClientFactory())->nationalCloud("https://www.microsoft.com");
+    }
+
+    function testCreateWithNoConfigReturnsDefaultClient() {
+        $client = (new HttpClientFactory())->create();
+        $this->assertInstanceOf(\GuzzleHttp\Client::class, $client);
+    }
+
+    function testCreateWithConfigCreatesClient() {
+        $config = [
+            "proxy" => "localhost:8000",
+            "verify" => false
+        ];
+        $client = (new HttpClientFactory())->clientConfig($config)

```

Comment:
What's your thoughts on making the methods and properties static? We don't need more than one copy of HttpClientFactory. It would require some refactoring. This code would look something like:

```php
$client = HttpClientFactory::clientConfig($config)
                                                       ->nationalCloud(NationalCloud::GERMANY)
                                                       ->create();
```

Reply:
Sounds good since we don't need more than one like you say.
Added in latest changes.

Reply:
Thanks for making it singleton.